misc/ls-extensions/modules/bal-shell-service/src/main/java/io/ballerina/shell/service/ConsoleOutCollector.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.ballerina.shell.service;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Collector to collect output passed in to the console.
+ *
+ * @since 2201.1.1
+ */
+public class ConsoleOutCollector extends OutputStream {
+    private List<String> lines;
+    private StringBuilder buffer;
+
+    public ConsoleOutCollector() {
+        this.lines = new ArrayList<>();
+        this.buffer = new StringBuilder();
+    }
+
+    @Override
+    public void write(int i) throws IOException {
+        if (i == '\n') {
+            lines.add(buffer.toString());
+            buffer = new StringBuilder();
+        } else {
+            buffer.append((char) i);
```

Comment:
If there is no specific reason shall we use a variable name instead of `i` ?

Reply:
Updated [3e2185a](https://github.com/ballerina-platform/ballerina-lang/pull/35871/commits/3e2185a179958fcdd0064a9dcc4d83b5dbfb5f68).